src/test/webviews/util.ts

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { EventEmitter } from 'vscode'
+import { WebviewApi } from 'vscode-webview'
+import { WebviewClient, WebviewClientAgent } from '../../webviews/client'
+import { VueWebview, VueWebviewPanel, VueWebviewView } from '../../webviews/main'
+import { mock } from '../utilities/mockito'
+
+/**
+ * Mock API construct intended for low-level testing of framework code.
+ *
+ * Test code that injects their own {@link WebviewClient} should not use this.
+ * Only use this class when manually responding to messages as sent by the client agent.
+ */
+export class MockWebviewApi<T, M = any> implements WebviewApi<T> {
```

Comment:
This class is not used much in this PR but it's still required since our webviews expect the API to just exist. Should be able to get rid of it if I change some other things.